compiler/rustc_hir/src/definitions.rs

```diff
@@ -202,7 +202,12 @@ impl DefPath {
         let mut s = String::with_capacity(self.data.len() * 16);
 
         for component in &self.data {
-            write!(s, "::{}[{}]", component.data.as_symbol(), component.disambiguator).unwrap();
+            match component.data.get_name() {
+                DefPathDataName::Named(name) => write!(s, "::{}", name).unwrap(),
```

Comment:
This was always printing the disambiguator so you should also print it with `#` for the `Named` case (but you can limit it to the case where `disambiguator != 0` if you want).